url.bs

```diff
@@ -2311,11 +2316,10 @@ background information. [[!HTML]]
 <dl class=switch>
  <dt>"<code>blob</code>"
  <dd>
-  <p>Let <var>url</var> be the result of <a lt="basic URL parser">parsing</a> <a for=/>URL</a>'s
-  <a for=url>path</a>[0].
+  If <a for=/>URL</a>'s <a for=url>cached origin</a> is not null, return
+  <a for=/>URL</a>'s <a for=url>cached origin</a>.
 
-  <p>Return a new <a>opaque origin</a>, if <var>url</var> is failure, and <var>url</var>'s
-  <a for=url>origin</a> otherwise.
+  <p>Return the result of <a for="blob URL" lt="resolve the origin">resolving the origin of the blob URL</a>.
   <!-- Did you mean: recursion -->
```

Comment:
There's no more recursion in your algorithm so this can be removed.

Reply:
Hmm I see now that https://w3c.github.io/FileAPI/#blob-url-resolve-the-origin still does recursion. I guess the advantage of that is making it less observable when things are revoked and allowing for a fast path.

Reply:
Yeah, making it less observable when things are revoked, and also making it work for URLs that were never created in the first place, or if some spec creates a URL record manually without going through the parsing steps.